Automattic/wp-calypso
# Themes: Add server route for logged-out /design

This PR renders our minimal `logged-out-design` layout on the `/design` route _on the server side_ (which currently just contains the masterbar). This will fix part of #1327 once done, and is based on #846.
It also enables that logged-out route on `wpcalypso`.

**Visual changes (logged-out calypso.localhost:3000/design, with JavaScript disabled):**

Before:

![bildschirmfoto am 2016-01-04 um 23 37 04](https://cloud.githubusercontent.com/assets/96308/12102592/3187feda-b33c-11e5-93a6-191fdb06759d.png)

After:

![bildschirmfoto am 2016-01-04 um 23 35 13](https://cloud.githubusercontent.com/assets/96308/12102560/e15db8f0-b33b-11e5-9fed-57a0a774307c.png)

**To test:**
* Restart Calypso, open an incognito browser window, disable JavaScript.
* Navigate to calypso.localhost:3000/design
* You should see the masterbar (see screenshot above)
* Re-enable JavaScript, reload.
* The complete (logged-out) theme showcase should load. *React tree reconciliation at work!* :tada:

TODO: see #1327
